pkgs/development/compilers/ponyc/default.nix

```
@@ -1,85 +1,115 @@
-{ stdenv, fetchFromGitHub, llvm, makeWrapper, pcre2, coreutils, which, libressl, libxml2,
-  cc ? stdenv.cc, lto ? !stdenv.isDarwin }:
+{ clangStdenv, fetchFromGitHub, fetchurl, llvm, makeWrapper, pcre2, coreutils, which, libressl, libxml2, cmake, z3, substituteAll,
+  cc ? clangStdenv.cc, lto ? !clangStdenv.isDarwin }:
 
-stdenv.mkDerivation ( rec {
+clangStdenv.mkDerivation ( rec {
   pname = "ponyc";
-  version = "0.33.2";
+  version = "0.38.1";
 
   src = fetchFromGitHub {
     owner = "ponylang";
     repo = pname;
     rev = version;
-    sha256 = "0jcdr1r3g8sm3q9fcc87d6x98fg581n6hb90hz7r08mzn4bwvysw";
+    sha256 = "1hk810k9h3bl641pgw91y4x2qw67rvbapx6p2pk9qz5p7nfcn7qh";
+
+# Due to a bug in LLVM 9.x, ponyc has to include its own vendored patched
+# LLVM.  (The submodule is a specific tag in the LLVM source tree).
+#
+# The pony developers are currently working to get off 9.x as quickly
+# as possible so hopefully in a few revisions this package build will
+# become a lot simpler.
+#
+# https://reviews.llvm.org/rG9f4f237e29e7150dfcf04ae78fa287d2dc8d48e2
+
+    fetchSubmodules = true;
   };
 
-  buildInputs = [ llvm makeWrapper which libxml2 ];
+# build time dependencies for the patched LLVM also need patches
+#  googletest = fetchurl {
+#    url = https://github.com/google/googletest/archive/release-1.8.1.tar.gz;
+#    sha256 = "17147961i01fl099ygxjx4asvjanwdd446nwbq9v8156h98zxwcv";
+#    name = "release-1.8.1.tar.gz";
+#  };
+  ponygbenchmark = fetchurl {
+    url = https://github.com/google/benchmark/archive/v1.5.0.tar.gz;
+    sha256 = "06i2cr4rj126m1zfz0x1rbxv1mw1l7a11mzal5kqk56cdrdicsiw";
+    name = "v1.5.0.tar.gz";
+  };
+
+
+  buildInputs = [ makeWrapper which libxml2 cmake z3 ];
   propagatedBuildInputs = [ cc ];
 
-  # Disable problematic networking tests
-  patches = [ ./disable-tests.patch ];
+  # Sandbox disallows network access, so disabling problematic networking tests
+  patches = [ ./disable-tests.patch
```

A small indentation issue here
```suggestion
  patches = [
    ./disable-tests.patch
    (substituteAll {
      src = ./make-safe-for-sandbox.patch;
      googletest = fetchurl {
        url = https://github.com/google/googletest/archive/release-1.8.1.tar.gz;
        sha256 = "17147961i01fl099ygxjx4asvjanwdd446nwbq9v8156h98zxwcv";
        name = "release-1.8.1.tar.gz";
      };
    })
  ];
```

---

Got it